devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java

```diff
@@ -66,38 +66,50 @@ public static void initConfigurations(Project project) {
         final ConfigurationContainer configContainer = project.getConfigurations();
 
         // Custom configuration for dev mode
-        configContainer.create(ToolingUtils.DEV_MODE_CONFIGURATION_NAME)
-                .extendsFrom(configContainer.getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+        configContainer.register(ToolingUtils.DEV_MODE_CONFIGURATION_NAME, config -> {
```

Comment:
`.register` w/ config action (and also `.named`) are nicer (less code to run if the Quarkus Gradle tasks aren't needed in a build - aka "Gradle's configuration on demand")